### PR TITLE
create stuc

### PR DESCRIPTION
**Pull request by nhatdaoquang25**

### CODE REVIEW

File: analysis_options.yaml [hunk 1]
```diff
@@ -21,7 +21,7 @@ linter:
   # or a specific dart file by using the `// ignore: name_of_lint` and
   # `// ignore_for_file: name_of_lint` syntax on the line or in the file
   # producing the lint.
-  rules:
+  rules:
```
Comment (thanhphamtrung) [inline analysis_options.yaml]: what change?

File: lib/src/app.dart [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      debugShowCheckedModeBanner: false,
+      initialRoute: constants.NameRoutesConstant.root,
+      onGenerateRoute: RouteGenerator.generateRoute,
+      onUnknownRoute: (RouteSettings settings) {
+        return MaterialPageRoute<void>(
+          settings: settings,
+          builder: (BuildContext context) =>
+              const Scaffold(body: Center(child: Text('Not Found'))),
```
Comment (thanhphamtrung) [inline lib/src/app.dart]: bring Not Found to constant

File: lib/src/config/constants.dart [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class AppConstants {
+  static const String ENV = 'ENV';
+  static const String HOST_NAME = 'HOST_NAME';
+  static const String DEFAULT_ENV = 'coin';
```
Comment (thanhphamtrung) [inline lib/src/config/constants.dart]: I think ENV should be Prod/Dev/Uat

File: lib/src/constants/constants.dart [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  static const String root = '/';
+  static const String detailScreen = '/detail';
+  static const String searchScreen = '/search';
+}
+
+class StringConstant {
+  static const String titleCoinScreen = 'COIN';
+  static const String titleDetailScreen = 'DETAIL SCREEN';
+  static const String titleSearchScreen = 'SEARCH SCREEN';
+}
```
Comment (thanhphamtrung) [inline lib/src/constants/constants.dart]: I think you should move it to separate

File: analysis_options.yaml (hunk 1 above)
Comment (nhatdaoquang25) [reply analysis_options.yaml]: It's nothing. I enter something  and forgot to erase space.

File: lib/src/app.dart (hunk 2 above)
Comment (nhatdaoquang25) [reply lib/src/app.dart]: Yah, I will bring it to constant

File: lib/src/constants/constants.dart (hunk 4 above)
Comment (nhatdaoquang25) [reply lib/src/constants/constants.dart]: I  think it's in a file can be manage to more carefully.

File: lib/src/config/constants.dart (hunk 3 above)
Comment (nhatdaoquang25) [reply lib/src/config/constants.dart]: You can explain ENV for the first or last or both. I will change it.

Comment (nhatdaoquang25) [reply lib/src/config/constants.dart]: Can I change to become this code?

Comment (nhatdaoquang25) [reply lib/src/config/constants.dart]: static const String ENV = 'PROD';

File: lib/src/constants/constants.dart (hunk 4 above)
Comment (HaiNguyenPV12) [reply lib/src/constants/constants.dart]: a constants file should only contain simple variables such as strings, numbers or other basic types. Importing packages in this file is highly discouraged and can lead to performance issues in the future.

**Review by thanhphamtrung — APPROVED** [review]

**Pull request merged**